group_project_v2/project_api.py

```diff
@@ -24,6 +24,36 @@
 DEFAULT_EXPIRATION_TIME = timedelta(seconds=10)
 
 
+# pylint:disable=too-many-instance-attributes
+class UserDetails(object):
+    def __init__(self, **kwargs):
+        self.id = kwargs.get('id', None)
+        self.email = kwargs.get('email', None)
+        self.username = kwargs.get('username', None)
+        self.uri = kwargs.get('uri', None)
+        self.first_name = kwargs.get('first_name', None)
+        self.last_name = kwargs.get('last_name', None)
+        self._full_name = kwargs.get('full_name', None)
+        self.gender = kwargs.get('gender', None)
+        self.avatar_url = kwargs.get('avatar_url', None)
+        self.city = kwargs.get('city', None)
+        self.country = kwargs.get('country', None)
+        self.is_active = kwargs.get('is_active', None)
+        self.level_of_education = kwargs.get('level_of_education', None)
+        self.organization = kwargs.get('organization', None)
+
+    @property
+    def full_name(self):
+        if self._full_name:
+            return self._full_name
+        parts = [self.first_name, self.last_name]
+        return " ".join([str(part) for part in parts if part is not None])
```

Comment:
I'd expect names to be `unicode` objects containing non-ASCII chraracters, so `str(part)` will often fail.  Is converting to `str` really necessary?  otherwise, you could simply go with

    return u" ".join(filter(None, parts))

which will also avoid adding a space at the beginning or end if one of the parts is the empty string.